web/gds-user-ui/src/components/BasicDetail/index.tsx

```diff
@@ -71,6 +83,37 @@ const BasicDetails: React.FC<BasicDetailProps> = () => {
     }
   };
 
+  useEffect(() => {
+    // now we can refresh the stepper data and even reload the page
+    if (isBasicStepDeleted && isAllFormDeleted) {
+      getCertificateStep();
+      getCertificateStep();
+      setShouldResetForm(true);
+      // window.location.reload();
+    }
+    if (isBasicStepDeleted && !isAllFormDeleted) {
```

Comment:
we do not need this check for the others, only the basic step need this check because when the user reset all the form, he is redirected to the basic form.
For the other form, I just check if that step is deleted from the state.